pkg/finder/finder_test.go

```diff
@@ -0,0 +1,93 @@
+package finder
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/getsops/sops/v3"
+	"github.com/getsops/sops/v3/age"
+	"github.com/getsops/sops/v3/keys"
+	"github.com/getsops/sops/v3/kms"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestValidSopsFiles(t *testing.T) {
+	testDir := "testdata"
+
+	expectedResults := map[string]sops.Metadata{
+		"testdata/valid_sops_files/encrypted.env":  {},
+		"testdata/valid_sops_files/encrypted.ini":  {},
+		"testdata/valid_sops_files/encrypted.yaml": {},
+		"testdata/valid_sops_files/encrypted.yml":  {},
+		"testdata/valid_sops_files/encrypted.json": {},
+	}
+
+	// Loop through files in the testdata directory.
+	files, err := SopsFiles(testDir)
+	assert.NoError(t, err)
+	assert.Equal(t, len(expectedResults), len(files))
+}
+
+func TestInvalidSopsFiles(t *testing.T) {
+	testDir := "testdata/invalid_sops_files"
+
+	expectedResults := map[string]sops.Metadata{}
+
+	// Loop through files in the testdata directory.
+	files, err := SopsFiles(testDir)
+	assert.NoError(t, err)
+	assert.Equal(t, len(expectedResults), len(files))
+}
+
+func TestGetKeys(t *testing.T) {
+	dummyArn := "arn:aws:kms:us-east-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"
+	dummyRole := "dummy-role"
+	dummyEncryptionContext := map[string]*string{"foo": aws.String("bar")}
+
+	age, err := age.MasterKeyFromRecipient("age1lzd99uklcjnc0e7d860axevet2cz99ce9pq6tzuzd05l5nr28ams36nvun")
+	kms := kms.NewMasterKey(dummyArn, dummyRole, dummyEncryptionContext)
+	assert.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		metadata sops.Metadata
+		expected []string
+	}{
+		{
+			name:     "Empty Metadata",
+			metadata: sops.Metadata{},
+			expected: []string{},
+		},
+		{
+			name: "Single Key",
+			metadata: sops.Metadata{
+				KeyGroups: []sops.KeyGroup{
+					[]keys.MasterKey{age},
+				},
+			},
+			expected: []string{"age1lzd99uklcjnc0e7d860axevet2cz99ce9pq6tzuzd05l5nr28ams36nvun"},
+		},
+		{
+			name: "Two Keys",
+			metadata: sops.Metadata{
+				KeyGroups: []sops.KeyGroup{
+					[]keys.MasterKey{age, kms},
+				},
+			},
+			expected: []string{"age1lzd99uklcjnc0e7d860axevet2cz99ce9pq6tzuzd05l5nr28ams36nvun",
+				"arn:aws:kms:us-east-2:111122223333:key/1234abcd-12ab-34cd-56ef-1234567890ab"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			got := GetKeys(tc.metadata)
+
+			// Compare slices
+			if len(got) != len(tc.expected) {
+				t.Errorf("Expected %d keys, got %d keys", len(tc.expected), len(got))
+				return
+			}
```

Comment:
This only tests that both slices have the same length, but does not compare their contents.

You could use

```go
assert.Equal(t, tc.expected, got)
```

here if you care about the exact order of elements, or

```go
assert.ElementsMatch(t, tc.expected, got)
```

if you don't.